src/components/codemirror-jsmd-mode.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-param-reassign */
+/* need to disable no-param-reassign b/c codemirror require mutating state */
+import CodeMirror from 'codemirror'
+
+CodeMirror.defineMode('jsmd', () => {
+  const jsMode = CodeMirror.getMode({}, { name: 'javascript' })
+  const mdMode = CodeMirror.getMode({}, { name: 'markdown' })
+  const plainMode = CodeMirror.getMode({}, { name: 'text/plain' })
+  const fetchMode = CodeMirror.getMode({}, { name: 'fetch' })
+
+  return {
+    startState: () => ({
+      localMode: plainMode,
+      localState: null,
+    }),
+    token: (stream, state) => {
+      let thisToken
+      if (stream.sol() && stream.match(/%%\s*js.*/, true)) {
+        thisToken = 'line-background-cm-jsmd-delim-line'
+        state.localMode = jsMode
+        state.localState = jsMode.startState()
+      } else if (stream.sol() && stream.match(/%%\s*md.*/, true)) {
+        thisToken = 'line-background-cm-jsmd-delim-line'
+        state.localMode = mdMode
+        state.localState = mdMode.startState()
+      } else if (stream.sol() && stream.match(/%%\s*fetch.*/, true)) {
+        thisToken = 'line-background-cm-jsmd-delim-line'
+        state.localMode = fetchMode
+        state.localState = fetchMode.startState()
+      } else if (stream.sol() && stream.match(/%%.*/, true)) {
+        // NB: THIS ELSE IF HAS TO COME LAST !!!!!
+        // any '%%' without a known chunk type just
+        // delimits the block without changing mode
+        thisToken = 'line-background-cm-jsmd-delim-line'
```

Comment:
Should it do that, or switch to a generic mode with no syntax highlighting?

Reply:
important point: a %% delim without a known chunk type will inherit the chunk type above. so this is valid:
```
%% js
var x = new Array(10).fill(1)
%%
y = x.map((v,i)=>i)
```

in this case, we add the token style for the delimiter line, but we don't need to change the mode

Reply:
I think for the empty chunk type, that makes sense, but for:

```
%% js
var x = 42
%% rust

```

Say, we don't know about rust -- we don't want it to be highlighted or treated as js do we?

Reply:
ah, gotcha @mdboom! great point and good catch!